### PR TITLE
[SPRINT-255][MOB-1529][Android SDK] Kotlin Serialization Module - Invoice

### PR DESCRIPTION
**Pull request by cjae**
## What did I do?
* Create an Invoice model using Kotlin serialization
* Create Unit test 
---

## Testing
* InvoiceTest.kt
---

## Deployment/Dependency Checklist:

* [ ] [List_out_any_new_ENV_vars]
* [ ] [List_out_any_back_end_dependencies]
---
## PR notes contain:
* [ ] Checklist for any deployment dependencies (such as new env vars, back-end dependencies, etc)
## PR Checklist:
* [ ] **I tested this**
* [x] I updated from the base branch (usually `latest`)
* [ ] I attached screenshot(s) (if there were UI changes)
* [x] I used the following PR title format: "[SPRINT-XX][fat-xx] BLAH"
  * Add "[EFS]" to the PR title if it was in the ticket title, e.g. "[SPRINT-XX][fat-xx][EFS]"
* [x] I looked over the Files Changed tab
* [ ] I consulted with the ticket's product owner (if necessary)
* [x] I requested a peer review

### CODE REVIEW

File: api/src/main/java/com/staxpayments/api/models/Invoice.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    val total: Double?,
+    val meta: JsonElement?,
+    val status: String?,
+    val url: String?,
+    val reminder: JsonElement?,
+    val schedule: JsonElement?,
+    val customer: Customer?,
+    val user: User?,
+    val files: JsonElement?,
+    @SerialName("child_transactions") val childTransactions: List<JsonElement>?,
```
Comment (thomasvidas) [inline api/src/main/java/com/staxpayments/api/models/Invoice.kt:19]: change the type to `JsonElement?` instead of `List<JsonElement>?`

**Review by thomasvidas — CHANGES_REQUESTED** [review]
One quick change, otherwise, this is good!

File: api/src/test/java/com/staxpayments/api/models/InvoiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        assertEquals(originalData, restored)
+    }
+
+    @OptIn(ExperimentalSerializationApi::class)
+    @Test(expected = MissingFieldException::class)
+    fun test_FailedSerializeBack() {
+        val restored = Json.decodeFromString<Invoice>(badString)
+        assertEquals(originalData, restored)
+    }
+}
```
Comment (muryno) [inline api/src/test/java/com/staxpayments/api/models/InvoiceTest.kt:79]: format your file using ktlint

File: api/src/main/java/com/staxpayments/api/models/Invoice.kt (hunk 1 above)
Comment (cjae) [reply api/src/main/java/com/staxpayments/api/models/Invoice.kt]: That was deliberate, I was actually waiting on this PR to be merged #83.
The type there is a Transaction type. I have updated it.

**Review by thomasvidas — APPROVED** [review]

**Pull request merged**